plugins/destination/plugin.go

```diff
@@ -3,54 +3,127 @@ package destination
 import (
 	"context"
 	"fmt"
+	"sync"
 	"time"
 
 	"github.com/cloudquery/plugin-sdk/schema"
 	"github.com/cloudquery/plugin-sdk/specs"
 	"github.com/rs/zerolog"
-	"golang.org/x/sync/errgroup"
+)
+
+type writerType int
+
+const (
+	unmanaged writerType = iota
+	managed
 )
 
 type NewClientFunc func(context.Context, zerolog.Logger, specs.Destination) (Client, error)
 
+type ManagedWriter interface {
+	PreWriteTableBatch(ctx context.Context, tables schema.Tables) error
+	WriteTableBatch(ctx context.Context, table *schema.Table, data [][]interface{}) error
+}
+
+type UnimplementedManagedWriter struct{}
```

Comment:
Why do we need an unimplemented managed writer?

Reply:
Answering my own question: probably to make it easier for destination plugins to implement the interface without implementing the managed writer for now.

Reply:
Follow-up comment: It seems like the need for `UnimplementedManagedWriter` is a consequence of the destination Plugin interface now needing to support two different types of writers. It also leads us to needing to check whether we're using a managed or unmanaged writer. Rather than overload the Plugin interface, maybe we should create a new ManagedPlugin interface? (This could also have a BasePlugin interface that gets used by both `Plugin` and `ManagedPlugin`)

Reply:
I already tried that route but it didn't work as NewUnmanaged plugin will mean a new Server and this is something we don't really want as we don't know ahead of time if a plugin is managed or unmanaged.

Reply:
Ah I see. Though on the other hand, a plugin will never need to be both managed and unmanaged, so it seems a bit confusing to keep them in one interface and force the use of `UnimplementedManagedWriter` / `UnimplementedWriter` on all plugins? We may not know initially if a plugin will benefit from being managed or not, but we can always start with managed first and then fall back to unmanaged if that turns out to not work for some reason